_posts/2020-11-08-why-and-how-to-write-good-commit-messages.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+title:  "How to and Why Write Good Commit Messages"
+date:   2020-11-08 12:54:04 -0700
+categories: commit messages overview
+---
+## The Purpose of Commit Messages
+
+When you want to save a snippet of code to your Git repository, a commit command is used. This will tell whatever program you're using to take a snapshot of your work so it can be referred to at a later time. Committing your work documents your progress as you work through and perfect your code. A caption is necessary for each of these snapshots so you know what was changed and why certain decisions were made. This is the purpose of a commit message. *A commit message is a concise statement detailing how and why changes were made in your code.*
+
+## The Importance of Writing Good Commit Messages
+
+#### A well written commit message is the best way to communicate changes in your code.
+
+##### This includes communicating to:
+  1. Yourself  
+    The way you write messages can help you remember why you did something when you go back and look at them weeks or months later.
+  2. Your collaborators  
+    Being as descriptive as possible in just a few words can help others understand the progress of your work without confusing them.
+
+Writing concise commit messages will make the log of your Git repository easier to navigate and the flow of a project comprehensible. Uncomplicated messages are crucial for teams to effectively collaborate in GitHub.
+
+#### Things to keep in mind when creating a Git commit message
+  1. Specify the type of commit. Examples of commit types includes a fix, a  feature or style update, a test, documentation, etc.
+  2. Always use the imperative tense, **not past tense!**
+  3. Use a capital letter in the first word of the subject line.
+  4. Subject and body of a commit need to be separate - use a blank line.
+  5. The subject line should not exceed 50 characters.
+  6. Wrap text of body manually at 72 characters.
+  7. If you have a hard time keeping it brief, you're probably trying to commit too many changes at once. Commit frequently to avoid this.
+  8. Aim to explain why and what you did, not how you did it.
+  9. Do not end your commit message with a period.
+
+## How to write a commit message
+
+#### A commit message should always be able to fill in the blank part of this sentence:
```

Comment:
I really like this part but, I want to suggest putting real examples of what you might commit. For example, "Add an extra line to ReadMe", "Remove extra spaces" etc.. just to make it a more applicable.